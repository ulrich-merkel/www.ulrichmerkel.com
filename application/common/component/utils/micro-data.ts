/**
 * Es6 module for creating mirco data attributes.
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2021
 */
import { isValidString } from '../../utils/string';

type ItemTypeAttributes = {
    itemScope: boolean;
    itemType: string;
};

/**
 * Get component properties for itemProp and itemType.
 *
 * @param {boolean} itemType - The itemType property if available
 * @returns {object|null} The component props for spreading
 */
export function getItemTypeAttributes(
    itemType?: string
): ItemTypeAttributes | null {
    return isValidString(itemType)
        ? {
              itemScope: true,
              itemType
          }
        : null;
}
