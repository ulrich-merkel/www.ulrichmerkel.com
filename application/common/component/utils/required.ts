/**
 * Es6 module for creating form input required attributes.
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2021
 */
import { isBoolean } from 'lodash';

type RequiredAttributes = {
    'aria-required': boolean;
    required: boolean;
};

/**
 * Get component properties for required form input elements.
 *
 * @param {boolean} required - Whether the input element is required or not
 * @returns {object|null} The component props for spreading
 */
export function getRequiredAttributes(
    required?: boolean
): RequiredAttributes | null {
    return isBoolean(required) && required
        ? {
              'aria-required': true,
              required: true
          }
        : null;
}
