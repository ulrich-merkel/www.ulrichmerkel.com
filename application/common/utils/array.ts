/**
 * Es6 module for handling array related utilities.
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2021
 */

/**
 * Check if given "input" is a valid array with at least one entry.
 *
 * @param {Array<*>} array - The array to be checked
 * @returns {boolean} Whether this is an array with some entries
 */
export function isValidArray(array?: any[]): boolean {
    return Array.isArray(array) && array.length > 0;
}
