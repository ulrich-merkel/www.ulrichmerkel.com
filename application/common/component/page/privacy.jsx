/* eslint-disable immutable/no-mutation */
/**
 * Es6 module for React Component.
 * Page components combine section components to the
 * overall page layout.
 *
 * @file
 * @module
 * @flow weak
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2016
 * @version 0.0.3
 *
 * @requires react
 * @requires prop-types
 * @requires react-helmet
 * @requires common/component/decorator/add-page-tracking
 * @requires common/component/decorator/add-content
 * @requires common/utils/content
 * @requires common/component/layout/main
 * @requires common/component/section/text
 *
 * @changelog
 + - 0.0.3 Moved to stateless function
 * - 0.0.2 Rewritten for es2015
 * - 0.0.1 Basic functions and structure
 */
import React from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';

import addPageTracking from '../decorator/add-page-tracking';
import addContent from '../decorator/add-content';
import { getContentSection } from '../../utils/content';
import LayoutMain from '../layout/main';
import SectionText from '../section/text';

/**
 * Function representing a component to return a single react child element.
 *
 * @function
 * @param {Object} [props] - The current component props
 * @returns {ReactElement} React component markup
 */
function PagePrivacy(props) {
    const { content } = props;
    const contentSection = getContentSection(content);

    return (
        <LayoutMain>
            <Helmet {...contentSection('head')} />
            <SectionText content={contentSection('section1')} isMain />
        </LayoutMain>
    );
}

/**
 * Validate props via React.PropTypes helpers.
 *
 * @static
 * @type {Object}
 * @property {Object} [content={}] - The component translation config
 */
PagePrivacy.propTypes = {
    content: PropTypes.objectOf(PropTypes.oneOfType([
        PropTypes.string,
        PropTypes.number,
        PropTypes.array,
        PropTypes.object
    ]))
};

/**
 * Set defaults if props aren't available.
 *
 * @static
 * @type {Object}
 * @see PagePrivacy.propTypes
 */
PagePrivacy.defaultProps = {
    content: {}
};

export default addPageTracking(addContent('PagePrivacy')(PagePrivacy));
