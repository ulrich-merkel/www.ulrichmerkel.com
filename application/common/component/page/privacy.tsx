/**
 * Es6 module for React Component.
 * Page components combine section components to the
 * overall page layout.
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2021
 */
import { default as React, FunctionComponent } from 'react';
import Helmet from 'react-helmet';

import { addPageTracking } from '../decorator/add-page-tracking';
import { addContent } from '../decorator/add-content';
import { getContentSection } from '../../utils/content';
import { LayoutMain } from '../layout/main';
import { SectionText } from '../section/text';

type Props = {
    content: {};
};

/**
 * Function representing a component to return a single react child element.
 *
 * @function
 * @param {object} [props] - The current component props
 * @returns {ReactElement} React component markup
 */
const Page: FunctionComponent<Props> = (props) => {
    const { content } = props;
    const contentSection = getContentSection(content);

    return (
        <LayoutMain>
            <Helmet {...contentSection('head')} />
            <SectionText content={contentSection('section1')} isMain />
        </LayoutMain>
    );
};

export const PagePrivacy = addPageTracking(addContent('PagePrivacy')(Page));
