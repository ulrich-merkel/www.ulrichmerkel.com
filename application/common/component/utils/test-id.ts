/**
 * Es6 module for creating test id related attributes
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2021
 */
import { isValidString } from '../../utils/string';

type TestIdAttributes = {
    'data-testid': string;
};

/**
 * Get component properties for testing.
 *
 * @param {string} testId - The test id to be added
 * @returns {object|null} The component props for spreading
 */
export function getTestIdAttributes(testId?: string): TestIdAttributes | null {
    return isValidString(testId)
        ? {
              'data-testid': testId
          }
        : null;
}
