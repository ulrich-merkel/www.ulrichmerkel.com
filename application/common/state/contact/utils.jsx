/**
 * Es6 module for Redux Architecture.
 * Simple helper component to add utility functions.
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2016
 * @version 0.0.1
 *
 * @see {@link http://redux.js.org/docs/recipes/reducers/ImmutableUpdatePatterns.html}
 *
 * @TODO Should be refactored and moved to actions
 *
 * @requires common/utils/validation
 *
 * @changelog
 * - 0.0.1 Basic functions and structure
 */
import { required, email } from '../../utils/validation';
import { isBrowser } from '../../utils/environment';

/**
 * Validation helper to check all inputs.
 *
 * @function
 * @param {object} [state={}] - The current redux contact state if given
 * @returns {object} The validation status for all inputs as object
 */
function validate(state = {}) {
    return {
        name: required(state.name),
        email: required(state.email) && email(state.email),
        subject: required(state.subject),
        message: required(state.message)
    };
}

/**
 * Validation single or all input element(s), depending on parameter input.
 *
 * @function
 * @param {object} state - The current redux contact state
 * @returns {boolean} The validation status
 */
function isValid(state) {
    const validated = validate(state);

    return !Object.keys(validated).some(function someKey(key) {
        return validated[key] !== undefined && !validated[key];
    });
}

function canSendForm(state) {
    const isStateValid = isValid(state);

    return isStateValid
        && state.pristine
        && !state.pending
        && isBrowser(state);
}

export {
    canSendForm,
    isValid,
    validate
};
