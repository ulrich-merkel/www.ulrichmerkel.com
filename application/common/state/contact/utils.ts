/**
 * Es6 module for Redux Architecture.
 * Simple helper component to add utility functions.
 *
 * @file
 * @module
 *
 * @author hello@ulrichmerkel.com (Ulrich Merkel), 2021
 *
 * @see {@link http://redux.js.org/docs/recipes/reducers/ImmutableUpdatePatterns.html}
 */
import { required, email } from '../../utils/validation';
import { isBrowser } from '../../utils/environment';
import { ContactFormType, ValidateType } from './types';

/**
 * Validation helper to check all inputs.
 *
 * @param {object} [contactForm] - The current redux contact state if given
 * @returns {object} The validation status for all inputs as object
 */
export function validate(contactForm: ContactFormType): ValidateType {
    return {
        name: required(contactForm?.name),
        email: required(contactForm?.email) && email(contactForm?.email),
        subject: required(contactForm?.subject),
        message: required(contactForm?.message)
    };
}

/**
 * Validation single or all input element(s), depending on parameter input.
 *
 * @param {object} [contactForm] - The current redux contact state
 * @returns {boolean} The validation status
 */
export function isValid(contactForm: ContactFormType): boolean {
    const validated = validate(contactForm);
console.log({validated, contactForm})
    return !Object.keys(validated).some(function someKey(key) {
        return validated[key] !== undefined && !validated[key];
    });
}

/**
 * Check if the form can be send.
 *
 * @param {object} [contactForm] - The current redux contact state
 * @returns {boolean} Whether the submit button is enabled or not
 */
export function canSendForm(contactForm: ContactFormType): boolean {
    const isStateValid = isValid(contactForm);

    return (
        isStateValid &&
        contactForm.pristine &&
        !contactForm.pending &&
        isBrowser(contactForm)
    );
}
