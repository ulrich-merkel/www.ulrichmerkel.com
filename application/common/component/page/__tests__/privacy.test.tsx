import * as React from 'react';
import { render } from '../../../../__tests__/utils/test-utils';
import { PagePrivacy } from '../privacy';

describe('PagePrivacy', function fnDescribe() {
    it('should render correctly', function fnIt() {
        const { asFragment } = render(<PagePrivacy />);
        expect(asFragment()).toMatchSnapshot();
    });
});
