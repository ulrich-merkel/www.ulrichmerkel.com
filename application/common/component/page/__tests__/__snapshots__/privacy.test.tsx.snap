// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PagePrivacy should render correctly 1`] = `
<DocumentFragment>
  <main
    class="l-main"
    itemprop="mainContentOfPage"
    role="main"
  >
    <section
      class="l-grid l-main__transition"
    >
      <div
        class="l-grid__spaced--small"
      >
        <div
          class="l-grid__row"
        >
          <div
            class="l-grid__col--12"
          >
            <article
              class="c-type--article is-main m-article"
              itemscope=""
              itemtype="https://schema.org/Article"
            >
              <h1
                class="is-centered c-type--h1 m-article__headline"
                itemprop="headline"
              >
                Privacy Statement
              </h1>
              <p
                class="is-centered c-type--p m-article__lead"
                itemprop="alternativeHeadline"
              >
                <strong>
                  Handling of your personal data
                </strong>
              </p>
              <div
                class="m-article__text"
                itemprop="text"
              >
                <div
                  class="m-text"
                >
                  <div
                    class="m-text__block"
                  >
                    <h3
                      class="is-centered c-type--h3 m-text__headline"
                    >
                      General
                    </h3>
                    <div
                      class="has-columns--2 m-text__content"
                    >
                      <p
                        class="c-type--p"
                      >
                        Your personal data (e.g. title, name, house address, e-mail address, phone number,
        bank details, credit card number) are processed by us only in accordance with
        the provisions of German data privacy laws. The following provisions describe
        the type, scope and purpose of collecting, processing and utilizing personal data.
        This data privacy policy applies only to our web pages. If links on our pages
        route you to other pages, please inquire there about how your data are handled
        in such cases.
                      </p>
                    </div>
                  </div>
                  <div
                    class="m-text__block"
                  >
                    <h3
                      class="is-centered c-type--h3 m-text__headline"
                    >
                      Inventory data
                    </h3>
                    <div
                      class="has-columns--2 m-text__content"
                    >
                      <p
                        class="c-type--p"
                      >
                        (1) Your personal data, insofar as these are necessary for this contractual relationship
        (inventory data) in terms of its establishment, organization of content and modifications,
        are used exclusively for fulfilling the contract. For goods to be delivered, for instance,
        your name and address must be relayed to the supplier of the goods.
                      </p>
                      <p
                        class="c-type--p"
                      >
                        (2) Without your explicit consent or a legal basis, your personal data are not passed on
        to third parties outside the scope of fulfilling this contract. After completion of the
        contract, your data are blocked against further use. After expiry of deadlines as per
        tax-related and commercial regulations, these data are deleted unless you have expressly
        consented to their further use.
                      </p>
                    </div>
                  </div>
                  <div
                    class="m-text__block"
                  >
                    <h3
                      class="is-centered c-type--h3 m-text__headline"
                    >
                      Web analysis with Google Analytics
                    </h3>
                    <div
                      class="has-columns--2 m-text__content"
                    >
                      <p
                        class="c-type--p"
                      >
                        This website uses Google Analytics, a web analysis service of Google Inc. (Google).
        Google Analytics uses cookies, i.e. text files stored on your computer to enable analysis
        of website usage by you. Information generated by the cookie about your use of this
        website is usually transmitted to a Google server in the United States and stored
        there. In case of activated IP anonymization on this website, however, your IP address
        is previously truncated by Google within member states of the European Union or in other
        states which are party to the agreement on the European Economic Area. Only in exceptional
        cases is a full IP address transmitted to a Google server in the United States and
        truncated there. On behalf this website's owner, Google will use this information to
        evaluate your use of the website, compile reports about website activities, and provide
        the website's operator with further services related to website and Internet usage.
        The IP address sent from your browser as part of Google Analytics is not merged with
        other data by Google. You can prevent storage of cookies by appropriately setting your
        browser software; in this case, however, please note that you might not be able to fully
        use all functions offered by this website. In addition, you can prevent data generated by
        the cookie and relating to your use of the website (including your IP address) from being
        collected and processed by Google, by downloading and installing a browser plug-in from
        the following link: 
                        <a
                          href="http://tools.google.com/dlpage/gaoptout?hl=en"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          http://tools.google.com/dlpage/gaoptout?hl=en
                        </a>
                      </p>
                      <p
                        class="c-type--p"
                      >
                        This website uses Google Analytics with the extension "anonymizeIP()", IP addresses being
        truncated before further processing in order to rule out direct associations to persons.
                      </p>
                    </div>
                  </div>
                  <div
                    class="m-text__block"
                  >
                    <h3
                      class="is-centered c-type--h3 m-text__headline"
                    >
                      Information about cookies
                    </h3>
                    <div
                      class="has-columns--2 m-text__content"
                    >
                      <p
                        class="c-type--p"
                      >
                        (1) To optimize our web presence, we use cookies. These are small text files stored in
        your computer's main memory. These cookies are deleted after you close the browser. Other
        cookies remain on your computer (long-term cookies) and permit its recognition on your
        next visit. This allows us to improve your access to our site.
                      </p>
                      <p
                        class="c-type--p"
                      >
                        (2) You can prevent storage of cookies by choosing a "disable cookies" option in your
        browser settings. But this can limit the functionality of our Internet offers as a result.
                      </p>
                    </div>
                  </div>
                  <div
                    class="m-text__block"
                  >
                    <h3
                      class="is-centered c-type--h3 m-text__headline"
                    >
                      Disclosure
                    </h3>
                    <div
                      class="has-columns--2 m-text__content"
                    >
                      <p
                        class="c-type--p"
                      >
                        According to the Federal Data Protection Act, you have a right to free-of-charge information
        about your stored data, and possibly entitlement to correction, blocking or deletion of such
        data. Inquiries can be directed to the following e-mail addresses:
        
                        <a
                          href="mailto:hello@ulrichmerkel.com"
                          title="Send email to hello@ulrichmerkel.com"
                        >
                          hello@ulrichmerkel.com
                        </a>
                      </p>
                    </div>
                  </div>
                </div>
              </div>
              <p
                class="is-centered c-type--p m-article__button hide-on-print"
              >
                <a
                  aria-current="page"
                  class="c-btn active"
                  href="/"
                  role="button"
                  title="Go to startpage"
                >
                  <span
                    class="c-btn__label"
                  >
                    Back to startpage
                  </span>
                </a>
              </p>
              <meta
                content="Privacy Statement"
                itemprop="name"
              />
              <meta
                content="2017-10-7"
                itemprop="datePublished"
              />
            </article>
          </div>
        </div>
      </div>
    </section>
  </main>
</DocumentFragment>
`;
